kurtosis-devnet/tests/eigenda/failover_test.go

```diff
@@ -0,0 +1,409 @@
+// The batches inbox transactions are queried via geth's GraphQL API.
+// TODO: We will also need to test the failover behavior of the node, which currently doesn't finalize after failover (fixed in https://github.com/Layr-Labs/optimism/pull/23)
+func TestFailover(t *testing.T) {
+	deadline, ok := t.Deadline()
+	if !ok {
+		deadline = time.Now().Add(1 * time.Minute)
+	}
+	ctxWithDeadline, cancel := context.WithDeadline(context.Background(), deadline)
+	defer cancel()
+
+	harness := newHarness(t)
+	t.Cleanup(func() {
+		// switch proxy back to normal mode, in case test gets cancelled
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		err := harness.clients.proxyMemconfigClient.Failback(ctx)
+		if err != nil {
+			t.Logf("Error failing back... you might need to reset proxy to normal mode manually: %v", err)
+		}
+	})
+
+	// assume kurtosis is running and is at least at block 10 (just deploying the contracts takes more than 10 blocks)
+	require.GreaterOrEqual(t, harness.testStartL1BlockNum, uint64(10), "Test started too early in the chain")
+	sinceBlock := harness.testStartL1BlockNum - 10
+
+	// 1. Check that the original commitments are EigenDA
+	harness.requireBatcherTxsToBeFromLayer(t, sinceBlock, DALayerEigenDA)
+
+	// 2. Failover and check that the commitments are now EthDA
+	t.Logf("Failover over... changing proxy's config to return 503 errors")
+	err := harness.clients.proxyMemconfigClient.Failover(ctxWithDeadline)
+	require.NoError(t, err)
+
+	afterFailoverL1BlockNum, err := harness.clients.gethL1Client.BlockNumber(ctxWithDeadline)
+	require.NoError(t, err)
+	// Wait for 10 L1 blocks. Assumption is that a cert is being posted every 2 blocks.
+	// Failover should take some time so we wait for 10 blocks to be sure some new commitment has started getting posted.
+	// TODO: read max-channel-duration from batcher's config instead of assuming 2 blocks
+	_, err = geth.WaitForBlock(big.NewInt(int64(afterFailoverL1BlockNum)+10), harness.clients.gethL1Client)
+	require.NoError(t, err)
+
+	harness.requireBatcherTxsToBeFromLayer(t, afterFailoverL1BlockNum, DALayerEth)
+
+	// 3. Failback and check that the commitments are EigenDA again
+	t.Logf("Failing back... changing proxy's config to start processing PUT requests normally again")
+	err = harness.clients.proxyMemconfigClient.Failback(ctxWithDeadline)
+	require.NoError(t, err)
+
+	afterFailbackL1BlockNum, err := harness.clients.gethL1Client.BlockNumber(ctxWithDeadline)
+	require.NoError(t, err)
+	_, err = geth.WaitForBlock(big.NewInt(int64(afterFailbackL1BlockNum)+10), harness.clients.gethL1Client)
+	require.NoError(t, err)
+
+	harness.requireBatcherTxsToBeFromLayer(t, afterFailbackL1BlockNum, DALayerEigenDA)
+
+}
+
+// Test Harness, which contains all the state needed to run the tests.
+// harness also defines some higher-level "require" methods that are used in the tests.
+type harness struct {
+	enclaveCtx          *enclaves.EnclaveContext
```

Comment:
not currently used. is it planned for future use, or just there as a template for future tests that might need it?